Description: Remove the hardcoded dependency on nautilus in most DEs, use xdg-utils
Author: Unit 193 <unit193@ninthfloor.org>

Forwarded: no
Last-Update: 2013-02-14

--- veracrypt.orig/src/Main/UserInterface.cpp
+++ veracrypt/src/Main/UserInterface.cpp
@@ -830,41 +830,13 @@ namespace VeraCrypt
 
 #else
 		// MIME handler for directory seems to be unavailable through wxWidgets
-		wxString desktop = GetTraits()->GetDesktopEnvironment();
-
-		if (desktop == L"GNOME" || desktop.empty())
-		{
-			args.push_back ("--no-default-window");
-			args.push_back ("--no-desktop");
-			args.push_back (string (path));
-			try
-			{
-				Process::Execute ("nautilus", args, 2000);
-			}
-			catch (TimeOut&) { }
-			catch (exception &e) { ShowError (e); }
-		}
-		else if (desktop == L"KDE")
+		args.push_back (string (path));
+		try
 		{
-			try
-			{
-				args.push_back (string (path));
-				Process::Execute ("dolphin", args, 2000);
-			}
-			catch (TimeOut&) { }
-			catch (exception&)
-			{
-				args.clear();
-				args.push_back ("openURL");
-				args.push_back (string (path));
-				try
-				{
-					Process::Execute ("kfmclient", args, 2000);
-				}
-				catch (TimeOut&) { }
-				catch (exception &e) { ShowError (e); }
-			}
+			Process::Execute ("xdg-open", args, 2000);
 		}
+		catch (TimeOut&) { }
+		catch (exception &e) { ShowError (e); }
 #endif
 	}
 
